crates/miden-lib/src/account/wallets/mod.rs

```diff
@@ -110,8 +110,9 @@ pub fn create_basic_wallet(
         ));
     }
 
-    let auth_component: AuthRpoFalcon512 = match auth_scheme {
-        AuthScheme::RpoFalcon512 { pub_key } => AuthRpoFalcon512::new(pub_key),
+    let auth_component: AccountComponent = match auth_scheme {
+        AuthScheme::RpoFalcon512 { pub_key } => AuthRpoFalcon512::new(pub_key).into(),
+        AuthScheme::NoAuth => NoAuth::new().into(),
```

Comment:
Same as above comment (and probably can be covered by the same issue): we should not be able to create a basic wallet with `NoAuth` component.